### DISCUSSION TITLE
Set background color for code output in HTML documents?

### DISCUSSION THREAD

**Discussion by maxdrohde** [quarto-dev/quarto-cli#2084] [Q&A]
I know that you can set the code block background color using the `code-block-bg` YAML option.

However, is there an option for setting the background color of the output part of the code block? I didn't see it documented on this page: https://quarto.org/docs/output-formats/html-code.html#appearance

**Comment by cscheid** [comment]
What do you mean by "output part"?

**Reply by maxdrohde** [reply]
I'm trying to set a background color for the HTML element that contains the results of the evaluated code -- like where the red box is

<img width="742" alt="Screen Shot 2022-08-23 at 17 10 37" src="https://user-images.githubusercontent.com/25734232/186281035-47030f27-ad9f-48fb-a85c-3d286f44d91a.png">

**Reply by cscheid** [reply]
If you stick to HTML, then the easiest thing is to just add CSS. The class is `cell-output`. So in your front matter, do

```
---
css: my-style.css
---
```

and then in `my-style.css`

```
div.cell-output {
    background-color: red;
}
```

ok maybe not red but you see the idea :)

**Comment by mcanouil** [comment]
I think this has already been answered (in some aspects) in <https://github.com/quarto-dev/quarto-cli/issues/1581> with a follow up to make it available not only with `knitr` engine (<https://github.com/quarto-dev/quarto-cli/issues/1581>).

<details><summary>See Quarto document content</summary>

````
---
title: "Untitled"
format: html
---

```{=html}
<style>
pre.source-code {
  background-color: #22b222;
}
pre.output-code {
  background-color: #b22222;
}
</style>
```

```{r}
#| class-source: source-code
#| class-output: output-code
1 + 1
```
````

</details>

<img width="707" alt="image" src="https://user-images.githubusercontent.com/8896044/186361388-dd6dab50-f56e-44b6-9a48-61aad895c664.png">

**Reply by cderv** [reply]
Thanks @mcanouil for linking issues. 

#1581 is a bit more specific IMO. You don't need such chunk options if you only want to style background color. I used that as example there because it was easier to show. 

Here, solution above using CSS seems the more generic as no SASS variable specific for this.

**Reply by mcanouil** [reply]
Well, it depends if the styling should be applied on the whole document or not though.
@cscheid example will overrides everything while the `class-*` allow for more fine-grained control, i.e., to highlight source/output differently for different context, etc.

@cderv I added this mostly as an alternative/reference.

**Reply by cderv** [reply]
Yes I agree, it is an alternative on for Quarto document with R when needing different style for different code output type. 

Generalization of this may happen in https://github.com/quarto-dev/quarto-cli/issues/2040 in the future.
